planner/core/rule_count_star_rewriter.go

```diff
@@ -0,0 +1,102 @@
+package core
+
+import (
+	"context"
+
+	"github.com/pingcap/tidb/expression"
+	"github.com/pingcap/tidb/expression/aggregation"
+	"github.com/pingcap/tidb/parser/mysql"
+)
+
+/**
+The countConstantRewriter is used to rewrite
+    count(constant value) -> count(not null column)
+Attention:
+Since count(*) is directly translated into count(1) during grammar parsing,
+the rewritten pattern also matches count(*)
+
+Pattern:
+LogcialAggregation
+       |
+   DataSource
+
+Optimize:
+Table
+  <k1 bool not null, k2 int null, k3 bigint not null>
+
+Case1 there are columns exists in datasource
+Query: select count(*) from table where k3=1
+CountStarRewriterRule: pick the narrowest not null column exists in datasource
+Rewritten Query: select count(k3) from table where k3=1
+
+Case2 there is no columns exists in datasource
+Query: select count(*) from table
+ColumnPruningRule: pick k1 as the narrowest not null column from origin table @Function.preferNotNullColumnFromTable
+                   datasource.columns: k1
+CountStarRewriterRule: rewrite count(*) -> count(k1)
```

Comment:
`CountStarRewriterRule`
No matter what name, the name should match the struct name

Reply:
forgot to change 😢